### DISCUSSION TITLE
getServerSideProps, There is a problem sending the request. Page stuck

### DISCUSSION THREAD

**Discussion by parlay96** [vercel/next.js#31621] [Help]
### What version of Next.js are you using?

12.0.2

### What version of Node.js are you using?

12.0.2

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying your application?

 next start

### Describe the Bug

![image](https://user-images.githubusercontent.com/32978177/142390245-968f6660-86b9-4c16-ab08-adc842ff9c5c.png)
Why is the server request page stuck and the client request normal

### Expected Behavior

![image](https://user-images.githubusercontent.com/32978177/142390245-968f6660-86b9-4c16-ab08-adc842ff9c5c.png)
Why is the server request page stuck and the client request normal

### To Reproduce

![image](https://user-images.githubusercontent.com/32978177/142390245-968f6660-86b9-4c16-ab08-adc842ff9c5c.png)
Why is the server request page stuck and the client request normal

**Comment by parlay96** [comment]
![image](https://user-images.githubusercontent.com/32978177/142390587-df7c855b-8996-4f13-be0e-4039f85decbb.png)
Page stuck

**Comment by parlay96** [comment]
![Uploading image.png…]()

**Comment by parlay96** [comment]
![image](https://user-images.githubusercontent.com/32978177/142390797-8b1c58aa-dbcd-4c5d-87cf-de0871917993.png)

**Comment by ljosberinn** [comment]
Please provide a reproduction, we don't know what your `fetch` does and very most likely its an issue on your requested resource.

**Comment by parlay96** [comment]
![image](https://user-images.githubusercontent.com/32978177/142401472-6f5694e8-c98b-47bc-a59b-53e3df4d4e1f.png)
I don't understand what this means at all

**Comment by parlay96** [comment]
![image](https://user-images.githubusercontent.com/32978177/142402861-2edf40d2-eb64-4aa0-a839-48aac1153456.png)
If the request time of this interface is very short, it is normal。
I did a simulation with the back end, and the data returned by the interface is all empty. Then the server request is normal

If the request from the back end of the interface takes a little more time, the rendering of the server will be stuck and the page will not be loaded

**Comment by parlay96** [comment]
@ljosberinn  If there is a problem with my request, how can I intercept it so that the page doesn't get stuck?

**Comment by parlay96** [comment]
@ljosberinn I found a problem and requested interface data. If the data is large, this problem will occur. Client does not exist

**Comment by ljosberinn** [comment]
> I don't understand what this means at all

It means exactly whats written there. Within `getServerSideProps`/`getStaticProps`, **do not** `fetch()` your `pages/api/*`.

> If the request time of this interface is very short, it is normal。
> I did a simulation with the back end, and the data returned by the interface is all empty. Then the server request is normal

> If the request from the back end of the interface takes a little more time, the rendering of the server will be stuck and the page will not be loaded

As far as I'm aware you cannot `await dispatch()`.

It appears the resource you're requesting is simply taking too long. Not an issue related to Next.js.

**Comment by parlay96** [comment]
Thank you. I think I know the problem

**Comment by parlay96** [comment]
@ljosberinn 
![image](https://user-images.githubusercontent.com/32978177/142551116-971d3017-3025-4085-9f7e-96573240e964.png)
I found that it was caused by the returned data clone on the node side
I use server-side rendering here. My request has an interceptor

**Comment by parlay96** [comment]
Only clone the response in the browser environment. If you clone the response in the node environment, there will be problems: https://github.com/bitinn/node-fetch/issues/553

**Comment by parlay96** [comment]
@ljosberinn As far as I'm aware you cannot await dispatch().？  At present, I have no problem using it like this. Can you tell me the reason?

**Comment by parlay96** [comment]
@ljosberinn 
![image](https://user-images.githubusercontent.com/32978177/142605623-88c04149-c190-4cc6-97c0-fa526cf02f05.png)
Why is my request interface adjusted twice for server rendering?？ I only sent it once,

**Comment by ljosberinn** [comment]
I believe dispatch is always sync and thus cannot be awaited.

Please don't tag me in unrelated questions to the nature of SSR in dev. This issue should be closed.
